### PR TITLE
Update login screens

### PR DESCRIPTION
**Pull request by zusoomro**
Plugs loading in the redux state into the Login and Register component. If the state.loading == true, show a spinner instead of the submit button to let the user know that their request is loading.

### CODE REVIEW

File: mobile/src/Register.tsx [hunk 1]
```diff
@@ -6,51 +6,61 @@ import {
...
-            <Button
-              accessibilityLabel="Submit"
-              onPress={handleSubmit}
-              title="Submit"
-            />
-          </View>
-        )}
-      </Formik>
-    </SafeAreaView>
+    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
```
Comment (allysmith12) [inline mobile/src/Register.tsx:23]: I tried searching on docs, but what exactly does this do?

**Review by allysmith12 — APPROVED** [review]
LGTM, just confused about the purpose of TouchableWithoutFeedback? ReactNative docs dont really explain.

File: mobile/src/Register.tsx (hunk 1 above)
Comment (zusoomro) [reply mobile/src/Register.tsx:23]: When you tap anywhere outside of the form fields, this dismisses the keyboard. The TouchableWithoutFeedback is basically a big invisible button that surrounds the component that captures the press event when it's not in the email or password field. This is so if you want to go to another page or scroll around or whatever, you can do so without the keyboard being in the way.

Comment (allysmith12) [reply mobile/src/Register.tsx:23]: I understand. https://reactnative.dev/docs/touchablewithoutfeedback/ says do not use TouchableWithoutFeedback unless you have a very good reason & to check out Pressable API for a " a more extensive and future-proof way to handle touch-based input". Maybe we can incorporate Pressable in the future!

Comment (zusoomro) [reply mobile/src/Register.tsx:23]: Sounds good. Yeah we don't really need any feedback on exiting out of the fields so this is what I'm used to using. The Pressable component looks like its more fully featured, so makes sense to use that in the future.

**Pull request merged**